engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetCullFace.java

```diff
@@ -55,45 +40,30 @@ public StateChange getDefaultInstance() {
     }
 
     @Override
-    public RenderPipelineTask generateTask() {
-        if (task == null) {
-            task = new SetCullFaceTask(mode);
+    protected RenderPipelineTask getDisablingTask() {
+        if (disablingTask == null) {
+            disablingTask = new DisableCapabilityTask(CAPABILITY);
         }
 
-        return task;
+        return disablingTask;
     }
 
     @Override
-    public boolean isEqualTo(StateChange stateChange) {
-        if (stateChange instanceof SetCullFace) {
-            return mode == ((SetCullFace) stateChange).getMode();
+    protected RenderPipelineTask getEnablingTask() {
```

Comment:
Agreed on slack to keep the methods but move null check and task instantiation in the constructor.